packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts

```diff
@@ -268,10 +270,39 @@ export class AmplifyAuthTransform extends AmplifyCategoryTransform {
     } else if (_.isEmpty(this._cognitoStackProps.triggers)) {
       parameters = Object.assign(parameters, { triggers: JSON.stringify(this._cognitoStackProps.triggers) });
     }
+
+    if (!this.validateCfnParameters(context, oldParameters, parameters)) {
+      process.exit(1);
+    }
+
     //save parameters
     JSONUtilities.writeJson(parametersJSONFilePath, parameters);
   };
 
+  public validateCfnParameters(context: $TSContext, oldParameters: $TSAny, parametersJson: $TSAny) {
```

Comment:
Since this prints things, it's probably not very reusable. Maybe rename this to `validateCfnParametersOrExit()` and move the `process.exit(1)` from above into here.

Reply:
I mostly did it this way to improve the ergonomics around testing it.

Reply:
Can you mock `process.exit()`?

Reply:
I can it's just not ergonomic

Reply:
I don't feel strongly, so I won't hold it up on this.